### ISSUE TITLE
`Unused local variable` drop automatically

### ISSUE THREAD

**Issue by vladyslavstartsev** [kalessil/phpinspectionsea#537]
drop automatically variable and assignment of it

**Comment by rentalhost** [comment]
I think that it is not possible. Inspections are only like "errors or improvements on code". You can create fixes that should be triggered manually, but not "automatically run it", once that it could be not expected for developer (for instance, he could expect to use this unused variable later). 

And drop the assignment could change the code behaviour. For instance:

```php
$i = 0;
$unused = ++$i;
echo $i;
```

It should print `1`, and if we drop the `$unused` with the assignment, it will print `0`.

**Comment by vladyslavstartsev** [comment]
@rentalhost then it should have a message that says - "apply at own risk" or something like that.
For `he could expect to use this unused variable later` - I think it's not the best idea to randomly leave unused variables in the code.
For 
```php
$i = 0;
$unused = ++$i;
echo $i;
```
then just drop the variable(`$unused`) and leave function call.

Update - check for increments/decrements, functions/methods that gets variable by reference, dynamic function calls (`$unused = $callSomething()`), [Variable variables](http://fr.php.net/manual/en/language.variables.variable.php)

But yes, it's kind of hard one.

**Comment by kalessil** [comment]
The plugin is not so intelligent =) Too many cases need to be considered, so I'd not provide QF for this.
